crates/pera-core/src/dwallet_mpc/dkg.rs

```diff
@@ -176,22 +212,17 @@ impl BytesParty for SecondDKGBytesParty {
         self,
         messages: HashMap<PartyID, Vec<u8>>,
         auxiliary_input: Vec<u8>,
-    ) -> Result<AdvanceResult, twopc_mpc::Error> {
+    ) -> PeraResult<AdvanceResult> {
         let auxiliary_input =
-            bcs::from_bytes(&auxiliary_input).map_err(|_| twopc_mpc::Error::InvalidParameters)?;
-        let messages = messages
-            .into_iter()
-            .map(|(k, v)| {
-                let message = bcs::from_bytes(&v).map_err(|_| twopc_mpc::Error::InvalidParameters);
-                return match message {
-                    Ok(message) => Ok((k, message)),
-                    Err(err) => Err(err),
-                };
-            })
-            .collect::<Result<HashMap<_, _>, _>>()?;
+            bcs::from_bytes(&auxiliary_input).map_err(|_| PeraError::DWalletMPCInvalidUserInput)?;
```

Comment:
Given that this is the second DKG round, are we sure this is again `InvalidUserInput`?

Reply:
Yeah. Also in this case, the auxiliary input deserialization will only fail if the user provided invalid input